server/pulp/server/webservices/views/lazy.py

```diff
@@ -0,0 +1,89 @@
+from django.http import HttpResponseRedirect, HttpResponseNotFound
+from django.views.generic import View
+
+from pulp.common.config import parse_bool
+from pulp.server.lazy.alias import AliasTable
+from pulp.server.lazy.url import URL, Key
+from pulp.server.config import config as pulp_conf
+
+
+class RedirectView(View):
+    """
+    The content redirect view redirects unsatisfied content
+    requests to the lazy streamer.
+
+    :ivar alias: An apache alias table.
+    :type alias: AliasTable
+    :ivar key: The public RSA key used for URL signing.
+    :type key: M2Crypto.RSA.RSA
+    """
+
+    @staticmethod
+    def urljoin(scheme, host, port, base, path, query):
```

Comment:
You'd think so (and I intended to) but `urlparse.urljoin()` makes really stupid assumptions about the parts that it is joining with respect to leading and trailing `/`.  Much like `os.path.join`.  So, by the time you *prepare* the url parts to be joined correctly by `urljoin()` you don't gain anything my using it.  We can talk about this unintuitive behavior sometime.